### PR TITLE
Get rid of MonadBlueprint and move functions from ArithmeticCircuit.Combinators to relevant places of usage

### PR DESCRIPTION
**Pull request by hovanja2011**

### CODE REVIEW

File: tests/Tests/ArithmeticCircuit.hs [hunk 1]
```diff
@@ -4,34 +4,35 @@
...
 import           ZkFold.Symbolic.Data.Eq
 import           ZkFold.Symbolic.Data.FieldElement
-import           ZkFold.Symbolic.MonadCircuit                           (Arithmetic)
 
 correctHom0 :: forall a . (Arithmetic a, Scale a a, Show a) => (forall b . Field b => b) -> Property
 correctHom0 f = let r = fromFieldElement f in withMaxSuccess 1 $ checkClosedCircuit r .&&. exec1 r === f @a
 
 correctHom1 :: forall a . (Arithmetic a, Scale a a, Show a) => (forall b . Field b => b -> b) -> a -> Property
-correctHom1 f x = let r = fromFieldElement $ f (FieldElement $ embed x) in checkClosedCircuit r .&&. exec1 r === f x
+correctHom1 f x = let r = fromFieldElement $ f (FieldElement $ embed @(ArithmeticCircuit a) $ Par1 x) in checkClosedCircuit r .&&. exec1 r === f x
```
Comment (TurtlePU) [inline tests/Tests/ArithmeticCircuit.hs:31]: I think we can replace these calls to `embed` with `fromConstant :: a -> FieldElement (ArithmeticCircuit a)`

File: src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs [hunk 2]
```diff
@@ -33,6 +36,9 @@ instance (Arithmetic a, Arbitrary a) => Arbitrary (ArithmeticCircuit a Par1) whe
         let ac = mempty { acInput = [1..k], acOutput = pure k }
         fromFieldElement <$> arbitrary' (FieldElement ac) 10
 
+getAllVars :: MultiplicativeMonoid a => ArithmeticCircuit a o -> [Natural]
```
Comment (TurtlePU) [inline src/ZkFold/Symbolic/Compiler/ArithmeticCircuit/Instance.hs:39]: Duplicate here and in `Map` module, consider moving to `Internal`

File: tests/Tests/ArithmeticCircuit.hs [hunk 3]
```diff
@@ -48,19 +49,19 @@ specArithmeticCircuit' = hspec $ do
...
         it "checks isZero(nonzero)" $ \(x :: a) ->
-          let Bool (r :: ArithmeticCircuit a Par1) = isZero $ FieldElement (embed x)
+          let Bool (r :: ArithmeticCircuit a Par1) = isZero $ FieldElement $ embed @(ArithmeticCircuit a) $ Par1 x
            in checkClosedCircuit r .&&. exec1 r === bool zero one (x Haskell.== zero)
         it "checks isZero(0)" $
           let Bool (r :: ArithmeticCircuit a Par1) = isZero (zero :: FieldElement (ArithmeticCircuit a))
            in withMaxSuccess 1 $ checkClosedCircuit r .&&. exec1 r === one
         it "computes binary expansion" $ \(x :: a) ->
-          let rs = binaryExpansion $ FieldElement (embed x)
+          let rs = binaryExpansion $ FieldElement $ embed @(ArithmeticCircuit a) $ Par1 x
```
Comment (TurtlePU) [inline tests/Tests/ArithmeticCircuit.hs:58]: here too, `FieldElement . embed` -> `fromConstant`

File: tests/Tests/ArithmeticCircuit.hs [hunk 4]
```diff
@@ -48,19 +49,19 @@ specArithmeticCircuit' = hspec $ do
         it "inverts nonzero correctly" $ correctHom1 @a finv
         it "inverts zero correctly" $ correctHom0 @a (finv zero)
         it "checks isZero(nonzero)" $ \(x :: a) ->
-          let Bool (r :: ArithmeticCircuit a Par1) = isZero $ FieldElement (embed x)
+          let Bool (r :: ArithmeticCircuit a Par1) = isZero $ FieldElement $ embed @(ArithmeticCircuit a) $ Par1 x
```
Comment (TurtlePU) [inline tests/Tests/ArithmeticCircuit.hs:52]: `FieldElement . embed . Par1` -> `fromConstant`

File: tests/Tests/ArithmeticCircuit.hs [hunk 5]
```diff
@@ -48,19 +49,19 @@ specArithmeticCircuit' = hspec $ do
...
         it "checks isZero(0)" $
           let Bool (r :: ArithmeticCircuit a Par1) = isZero (zero :: FieldElement (ArithmeticCircuit a))
            in withMaxSuccess 1 $ checkClosedCircuit r .&&. exec1 r === one
         it "computes binary expansion" $ \(x :: a) ->
-          let rs = binaryExpansion $ FieldElement (embed x)
+          let rs = binaryExpansion $ FieldElement $ embed @(ArithmeticCircuit a) $ Par1 x
            in checkClosedCircuit rs .&&. V.fromVector (exec rs) === padBits (numberOfBits @a) (binaryExpansion x)
         it "internalizes equality" $ \(x :: a) (y :: a) ->
-          let Bool (r :: ArithmeticCircuit a Par1) = embed x == embed y
+          let Bool (r :: ArithmeticCircuit a Par1) = embed @(ArithmeticCircuit a) (Par1 x) == embed @(ArithmeticCircuit a) (Par1 y)
```
Comment (TurtlePU) [inline tests/Tests/ArithmeticCircuit.hs:61]: better to check equality on `FieldElements` with `fromConstant`

File: tests/Tests/ArithmeticCircuit.hs [hunk 6]
```diff
@@ -48,19 +49,19 @@ specArithmeticCircuit' = hspec $ do
...
            in checkClosedCircuit r .&&. exec1 r === bool zero one (x Haskell.== y)
         it "internal equality is reflexive" $ \(x :: a) ->
-          let Bool (r :: ArithmeticCircuit a Par1) = embed x == embed x
+          let Bool (r :: ArithmeticCircuit a Par1) = embed @(ArithmeticCircuit a) (Par1 x) == embed (Par1 x)
```
Comment (TurtlePU) [inline tests/Tests/ArithmeticCircuit.hs:64]: better to check equality on `FieldElement`s with `fromConstant`

**Review by TurtlePU — APPROVED** [review]

**Review by vlasin — APPROVED** [review]

**Pull request merged**